app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -143,16 +108,7 @@ object FirebaseDB {
         }
     }
 
-    /**
-     * Store the profile picture for the given [User]
-     *
-     * @param image the profile picture to store
-     * @param userId the [User] for whom to add the pictures
-     * @return the updated [PhotoMetadata] with the current id of the image we stored
-     *
-     * @throws IllegalArgumentException if the [userId] was blank or empty
-     */
-    suspend fun storeUserProfilePicture(image: Bitmap, userId: String, metadata: PhotoMetadata): PhotoMetadata? {
+    override suspend fun storeUserProfilePicture(image: Bitmap, userId: String, metadata: PhotoMetadata): PhotoMetadata? {
         if (userId.isBlank())

```

Comment:
A lot of code duplication here

Reply:
I am sorry, but I do not really see about what duplication you are actually speaking of ?